SU2_CFD/include/numerics_simd/CNumericsSIMD.hpp

```diff
@@ -0,0 +1,594 @@
+  for (size_t iDim = 0; iDim < nDim; ++iDim) {
+    jac(nDim+1,iDim+1) = scale * (normal(iDim)*a1 - gamma_m_1*velocity[iDim]*projVel);
+  }
+  jac(nDim+1,nDim+1) = scale * gamma * projVel;
+
+  return jac;
+}
+
+/*!
+ * \class CRoeScheme
+ * \brief Classic Roe scheme, with or without viscous terms.
+ */
+template<size_t nDim, bool VISCOUS>
+class CRoeScheme final : public CFinalLink<nDim> {
+private:
+  const Double kappa;
+  const Double gamma;
+  const Double entropyFix;
+  const bool dynamicGrid;
+
+public:
+  CRoeScheme(const CConfig& config) :
+    kappa(config.GetRoe_Kappa()),
+    gamma(config.GetGamma()),
+    entropyFix(config.GetEntropyFix_Coeff())
+    dynamicGrid(config.GetDynamic_Grid()) {
+
+  }
+
+  void ComputeFlux(unsigned long iEdge,
+                   const CConfig& config,
+                   const CGeometry& geometry,
+                   const CVariable& solution_,
+                   UpdateType updateType,
+                   CSysVector& vector,
+                   CSysMatrix& matrix) override {
+
+    const bool implicit = (config->GetKind_TimeIntScheme() == EULER_IMPLICIT);
+    const auto solution = static_cast<CEulerVariable&>(solution_);
+
+    constexpr size_t nVar = CCompressibleConservatives<nDim>::nVar;
+
+    const auto iPosize_t = geometry.edges->GetNodes(iEdge,0);
+    const auto jPosize_t = geometry.edges->GetNodes(iEdge,1);
+
+    /*--- Geometrix properties. ---*/
+
+    const auto vector_ij = distanceVector(iPosize_t, jPosize_t, geometry.nodes->Coord);
+
+    const auto normal = loadVariables<nDim>(iEdge, geometry.edges->Normal);
+    const auto area = norm(normal);
+    VectorDbl<nDim> unitNormal;
+    for (size_t iDim = 0; iDim < nDim; ++iDim) {
+      unitNormal(iDim) = normal(iDim) / area;
+    }
+
+    /*--- Reconstructed primitives. ---*/
+
+    auto V = getCompressiblePrimitives(iPosize_t, jPosize_t,
+                                       config.GetKind_SlopeLimit_Flow(),
+                                       vector_ij, solution);
+
+    /*--- Compute conservative variables. ---*/
+
+    CPair<CCompressibleConservatives<nDim> > U;
+    U.i = getCompressibleConservatives(V.i);
+    U.j = getCompressibleConservatives(V.j);
+
+    /*--- Roe averaged variables. ---*/
+
+    auto R = sqrt(abs(V.j.density / V.i.density));
+    auto densityRoe = R * V.i.density;
+    VectorDbl<nDim> velocityRoe;
+    for (size_t iDim = 0; iDim < nDim; ++iDim) {
+      velocityRoe(iDim) = (R*V.j.velocity(iDim) + V.i.velocity(iDim)) / (R+1);
+    }
+    auto enthalpyRoe = (R*V.j.enthalpy + V.i.enthalpy) / (R+1);
+    auto soundSpeedRoe = sqrt((gamma-1) * (enthalpyRoe - 0.5*squaredNorm(velocityRoe)));
+
+    /*--- P tensor and inverse P tensor. ---*/
+
+    auto projVelRoe = dot(velocityRoe, unitNormal);
+    auto pMat = pMatrix(gamma, densityRoe, velocityRoe, projVelRoe, soundSpeedRoe, unitNormal);
+    auto pMatInv = pMatrixInv(gamma, densityRoe, velocityRoe, projVelRoe, soundSpeedRoe, unitNormal);
```

Comment:
This does not compile yet, but in theory the vectorized code will look exactly the same (well it will look better because that's what hindsight tends to do).